### PR TITLE
i18n(de): recreate `basics/layout.mdx`

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description (required)

- delete the whole content because it was too out of date
- translate the english content from scratch

#### Related issues & labels (optional)

in theory this should update the translation of these PRs, but I think there are many more changes not tracked by Lunaria...
- #8763 
- #9333
- #9335 
- #9344 

(https://github.com/withastro/docs/commits/main/src/content/docs/en/basics/layouts.mdx?since=2024-07-10T18:34:38.000Z)

### CODE REVIEW

File: src/content/docs/de/basics/layouts.mdx [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
 
-Layouts werden normalerweise im Verzeichnis `src/layouts` deines Projekts abgelegt.
+Wenn eine Layoutkomponente jedoch eine Seitenhülle enthält, muss ihr `<html>`-Element das übergeordnete Element aller anderen Elemente der Komponente sein. Alle [`<style>`](/de/guides/styling/#styles-stile-in-astro) oder [`<script>`](/de/guides/client-side-scripts/#verwendung-von-script-in-astro) Elemente müssen von den `<html>` Tags umschlossen sein.
+
+Layout-Komponenten werden normalerweise in einem `src/layouts`-Verzeichnis in deinem Projekt platziert, aber das ist keine Voraussetzung; du kannst sie überall in deinem Projekt platzieren. Du kannst Layout-Komponenten sogar neben deinen Seiten platzieren, [indem du den Layout-Namen ein `_` voranstellst](/de/guides/routing/).
```
Comment (jsparkdev) [inline src/content/docs/de/basics/layouts.mdx:19]:
```suggestion
Layout-Komponenten werden normalerweise in einem `src/layouts`-Verzeichnis in deinem Projekt platziert, aber das ist keine Voraussetzung; du kannst sie überall in deinem Projekt platzieren. Du kannst Layout-Komponenten sogar neben deinen Seiten platzieren, [indem du den Layout-Namen ein `_` voranstellst](/de/guides/routing/#excluding-pages). 
```

The link seems to be different. If there is no heading that corresponds to the link, it seems like a good idea to keep the original link.

Comment (trueberryless) [reply src/content/docs/de/basics/layouts.mdx:19]: My thought process was:

- If there is no german file, link to the english heading.
- If there is a finished translation, link to the same locale heading.

But in this case, there is an outdated german version where this heading doesn't even exist yet... So linking to either the german or english heading will always land on top of the german page.

Is it now really better to link to the english heading, or should I already link to the german one (which I will translate in the future)?

Comment (jsparkdev) [reply src/content/docs/de/basics/layouts.mdx:19]: It's okay to keep it, it's just a suggestion.

The reason for the suggestion is that if you update the `routing.mdx` file after this PR is merged, you may forget that you need to update this file again.

If you add an English version of the heading, CI might throw an error when you update the `routing.mdx` file and realize that you need to update this file along with it.

Comment (trueberryless) [reply src/content/docs/de/basics/layouts.mdx:19]: Okay, this is a very inspiring way of keeping such things in mind! I wrote a TODO comment below this line, but I think I'll delete the TODO comment and update it. Thank you!

**Comment by trueberryless** [comment]
I guess the CI check failing is okay, as mention [here](https://github.com/withastro/docs/pull/10034#discussion_r1850219476), @jsparkdev ?

**Comment by sarah11918** [comment]
Just mentioning that this cannot merge with a failing check, because then all of docs will show a failing check because of this on the main branch itself, and it will affect everything!

**I would say to remove the link entirely**. The content doesn't exist on that page so it's not even a helpful link at all.

These docs are going to be retired soon, and almost all the new docs will be outdated anyway, so you'll probably HAVE to revisit this page. And, presumably at some point someone will update the routing page to actually include this heading, and at that point, the link can be readded.

Once v5 is published, this would be great to tackle: first update the routing page and then come back and add this link!

**Comment by jsparkdev** [comment]
@trueberryless @sarah11918 Thanks for the nice comment, I guess my suggestion wasn't a good one 😂

**Review by jsparkdev — APPROVED** [review]
LGTM!

**Pull request merged**